From d1126bcc9ab58062fd2d340a7011df9ebaf7f53b Mon Sep 17 00:00:00 2001
From: mtk27745 <rex.lu@mediatek.com>
Date: Mon, 6 Nov 2023 10:16:34 +0800
Subject: [PATCH 004/119] mtk: mt76: mt7996: Remove wed rro ring add napi at
 init state

without this patch. rro ring will add napi at initial state. once rro ring add napi, it will have chance to be used by host driver. if host driver accessed the ring data, it will cause some issue.

Signed-off-by: mtk27745 <rex.lu@mediatek.com>
---
 dma.c | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/dma.c b/dma.c
index 54d2dbc3..c132e37f 100644
--- a/dma.c
+++ b/dma.c
@@ -958,6 +958,10 @@ mt76_dma_init(struct mt76_dev *dev,
 	init_completion(&dev->mmio.wed_reset_complete);
 
 	mt76_for_each_q_rx(dev, i) {
+		if (mtk_wed_device_active(&dev->mmio.wed) &&
+		    mt76_queue_is_wed_rro(&dev->q_rx[i]))
+			continue;
+
 		netif_napi_add(&dev->napi_dev, &dev->napi[i], poll);
 		mt76_dma_rx_fill_buf(dev, &dev->q_rx[i], false);
 		napi_enable(&dev->napi[i]);
-- 
2.45.2

